LokiKit/ProofOfWork.swift

```diff
@@ -0,0 +1,156 @@
+import CryptoSwift
+
+private extension Int {
+    init(_ decimal: Decimal) {
+        let double = NSDecimalNumber(decimal: decimal).doubleValue
+        self.init(double)
+    }
+}
+
+private extension UInt8 {
+    init(_ decimal: Decimal) {
+        self.init(Int(decimal))
+    }
+}
+
+private extension Decimal {
+    /// Get the remainder of a Decimal
+    static func %(lhs: Decimal, rhs: Int) -> Decimal {
+        return Decimal(Int(lhs) % rhs)
+    }
+    
+    /// Divide a Decimal by an Int
+    static func /(lhs: Decimal, rhs: Int) -> Decimal {
+        return lhs / Decimal(rhs)
+    }
+
+    /// Convert a Decimal to a UInt8 array of a given length
+    func toArray(ofLength length: Int) -> [UInt8] {
+        return (0..<length).map { i in
+            let n = length - (i + 1)
+            // 256 ** n is the value of one bit in arr[i], modulus to carry over
+            // (self / 256**n) % 256;
+            let denominator = pow(256, n)
+            let fraction = self / denominator
+            
+            // fraction % 256
+            let remainder = fraction % 256
+            return UInt8(remainder)
+        }
+    }
+}
+
+// UInt8 Array specific stuff we need
+private extension Array where Element == UInt8 {
+    
+    /// Compare if lhs array is greater than rhs array
+    static func >(lhs: [UInt8], rhs: [UInt8]) -> Bool {
+        guard lhs.count == rhs.count else { return false }
+        
+        // lhs is greater than rhs if any value in lhs is greater than the corresponding value in the rhs
+        return zip(lhs, rhs).contains { $0 > $1 }
+    }
+    
+    
+    /// Increment the UInt8 array by a given amount
+    ///
+    /// - Parameter amount: The amount to increment by
+    /// - Returns: The incrememnted array
+    func increment(by amount: Int) -> [UInt8] {
+        var newNonce = [UInt8](self)
```

Comment:
Since arrays are value types in Swift you should be able to just do `var newNonce = self`